### PR TITLE
Create encryptwallet modal

### PR DESCRIPTION
**Pull request by kewde**

### CODE REVIEW

File: src/app/modals/encryptwallet/encryptwallet.component.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    </div>
+
+  </div>
+</div>
+
+<hr class="margin-top-100px">
+
+<app-password
+  #passwordElement
+  [buttonText]="'ENCRYPT WALLET'"
```
Comment (rynomster) [inline src/app/modals/encryptwallet/encryptwallet.component.html]: Change to
`buttonText="ENCRYPT WALLET"`

File: src/app/modals/encryptwallet/encryptwallet.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export class EncryptwalletComponent implements OnInit {
+
+  log: any = Log.create('unlockwallet.component');
+  public password: string;
+
+  @ViewChild('passwordElement') passwordElement: PasswordComponent;
+
+  constructor(private _rpc: RPCService) { }
+
+  ngOnInit() {
```
Comment (rynomster) [inline src/app/modals/encryptwallet/encryptwallet.component.ts]: Not used

File: src/app/modals/encryptwallet/encryptwallet.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    if (this.password) {
+
+      this.log.d(`check password equality: ${password.password === this.password}`);
+
+      if (this.password === password.password) {
+
+        this.log.d(`Encrypting wallet! password: ${this.password}`);
+        this._rpc.call('encryptwallet', [password.password])
+            .subscribe(
+            (response: any) => {
```
Comment (rynomster) [inline src/app/modals/encryptwallet/encryptwallet.component.ts]: Avoid alerts at all cost

File: src/app/modals/encryptwallet/encryptwallet.component.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+              alert(response);
+              document.getElementById('close').click();
```
Comment (rynomster) [inline src/app/modals/encryptwallet/encryptwallet.component.ts]: Avoid this, use refs

File: src/app/modals/encryptwallet/encryptwallet.component.html (hunk 1 above)
Comment (kewde) [reply src/app/modals/encryptwallet/encryptwallet.component.html]: You'd think that would work right, but I tried it a few months ago and it didn't for some odd reason.
Will check.

File: src/app/modals/encryptwallet/encryptwallet.component.ts (hunk 2 above)
Comment (kewde) [reply src/app/modals/encryptwallet/encryptwallet.component.ts]: generated by default

File: src/app/modals/encryptwallet/encryptwallet.component.ts (hunk 4 above)
Comment (kewde) [reply src/app/modals/encryptwallet/encryptwallet.component.ts]: Close is part of the modal, I can't inject the modalservice without circular dependencies..

**Comment by rynomster** [comment]
I meant the alert box.. Avoid alerts at all costs

**Comment by rynomster** [comment]
Still missing changes, and doesn't pass checks

File: src/app/modals/encryptwallet/encryptwallet.component.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            } ,
+            // Handle error appropriately
+            error => { 
+            	alert('Wallet failed to encrypt properly!');
```
Comment (rynomster) [inline src/app/modals/encryptwallet/encryptwallet.component.ts]: Please get rid of alert box.. This should use a modal. Alert boxes are blocking, and stops everything.

File: src/app/modals/encryptwallet/encryptwallet.component.html (hunk 1 above)
Comment (kewde) [reply src/app/modals/encryptwallet/encryptwallet.component.html]: Checked up on this, ' ' is required.

File: src/app/modals/encryptwallet/encryptwallet.component.ts (hunk 3 above)
Comment (kewde) [reply src/app/modals/encryptwallet/encryptwallet.component.ts]: replaced.

File: src/app/modals/encryptwallet/encryptwallet.component.html (hunk 1 above)
Comment (rynomster) [reply src/app/modals/encryptwallet/encryptwallet.component.html]: Have a look at createwallet.component.html :)

**Pull request merged**